pkg/kubernetes/objectmeta/v1alpha1/objectmeta.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2019 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/pkg/errors"
+
+	stringer "github.com/openebs/maya/pkg/apis/stringer/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ObjectMeta is a wrapper over metav1.ObjectMeta
+type ObjectMeta struct {
+	Object *metav1.ObjectMeta
+	errors []error
+}
+
+// String implements Stringer interface
+func (om ObjectMeta) String() string {
+	return stringer.Yaml("objectmeta", om)
+}
+
+// GoString implements GoStringer interface
+func (om ObjectMeta) GoString() string {
+	return om.String()
+}
+
+// New returns a new instance of ObjectMeta
+func New() *ObjectMeta {
+	return &ObjectMeta{
+		Object: &metav1.ObjectMeta{},
+	}
+}
+
+// WithName adds name in ObjectMeta instance
+func (om *ObjectMeta) WithName(name string) *ObjectMeta {
```

Comment:
Let us make use of `Builder` as the struct that does all these.
It will help us a lot in maintaining the standard.
More than standard it make the code very explicit & understood by new contributors.

Reply:
done PTAL.